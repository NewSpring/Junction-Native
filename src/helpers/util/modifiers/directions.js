import cap from "../upperCaseFirstLetter";
import lens from "../lenses/baseSize";

// generic direction reducer
const reducer = properties =>
  properties.map(({ property, props, name }) => ({
    [`${property}${cap(name)}`]: lens(property, props)[property],
  }));

// direction reducer for sides/ends (two results)
const doubleReducer = properties =>
  properties.map(({ property, props, name }) => {
    const names = name === "ends" ? ["Top", "Bottom"] : ["Left", "Right"];
    return {
      [`${property}${names[0]}`]: lens(property, props)[property],
      [`${property}${names[1]}`]: lens(property, props)[property],
    };
  });

export default [
  { name: "left", reducer },
  { name: "top", reducer },
  { name: "bottom", reducer },
  { name: "right", reducer },
  { name: "sides", reducer: doubleReducer },
  { name: "ends", reducer: doubleReducer },
];
