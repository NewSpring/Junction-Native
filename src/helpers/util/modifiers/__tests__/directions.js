import directions from "../directions";

it("should match the snapshot", () => {
  expect(directions).toMatchSnapshot();
});
