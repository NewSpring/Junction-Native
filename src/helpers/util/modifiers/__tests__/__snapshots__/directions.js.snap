// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the snapshot 1`] = `
Array [
  Object {
    "name": "left",
    "reducer": [Function],
  },
  Object {
    "name": "top",
    "reducer": [Function],
  },
  Object {
    "name": "bottom",
    "reducer": [Function],
  },
  Object {
    "name": "right",
    "reducer": [Function],
  },
  Object {
    "name": "sides",
    "reducer": [Function],
  },
  Object {
    "name": "ends",
    "reducer": [Function],
  },
]
`;
